docs/src/modules/java/pages/call-another-as-service.adoc

```diff
@@ -0,0 +1,118 @@
+
+The build needs to be updated to generate Akka gRPC client classes. In the existing configuration for the `protobuf-maven-plugin` in your `pom.xml` add the following:
+
+---
+// FIXME this block is a bit unwieldy, but har to understand where/what to add if not
+// FIXME should we _always_ include something instead, like a src/main/ex-proto and avoid clients for the service itself?
+```xml
+<plugin>
+  <groupId>org.xolstice.maven.plugins</groupId>
+  <artifactId>protobuf-maven-plugin</artifactId>
+  <version>0.6.1</version>
+    <executions>
+      <execution>
+        <!-- generate Java code for protobuf types -->
+        <id>protobuf</id>
+        <goals>
+          <goal>compile</goal>
+        </goals>
+        <configuration>
+          <writeDescriptorSet>true</writeDescriptorSet>
+          <includeSourceInfoInDescriptorSet>true</includeSourceInfoInDescriptorSet>
+          <descriptorSetFileName>user-function.desc</descriptorSetFileName>
+          <checkStaleness>true</checkStaleness>
+          <protocArtifact>com.google.protobuf:protoc:3.9.1:exe:${os.detected.classifier}</protocArtifact>
+          <!-- add this block --> // <1>
+          <protocPlugins>
+            <protocPlugin>
+              <id>akka-grpc</id>
+              <groupId>com.lightbend.akka.grpc</groupId>
+              <artifactId>akka-grpc-scalapb-protoc-plugin_2.12</artifactId>
+              <version>${akka-grpc.version}</version>
+              <mainClass>akka.grpc.gen.Main</mainClass>
+            </protocPlugin>
+          </protocPlugins>
+          <!-- end of block to add -->
+        </configuration>
+      </execution>
+   </executions>
+</plugin>
+```
+---
+<1> The `protocPlugins` block to add
+
+The Akka gRPC `protocPlugin` will now generate Akka gRPC client classes for the service when we compile it.
+
+To use the generated client we need to manually create an instance for it:
+
+// FIXME this leaks clients as a Action is created on each call, we either need to inject it and share or create a new one and close it for every call (but I think we'd want to re-use rather?)
+// FIXME withTls(false) doesn't seem to work
+// FIXME host/port should probably come from config? Or do we keep sample brief and just mention that instead?
+---
+```java
+/** An action. */
+public class ProxyAction extends AbstractProxyAction {
+
+  private final CounterServiceClient counterClient;
+
+  public ProxyAction(ActorSystem system, ActionCreationContext creationContext) { // <1>
+    // Configure the client by code:
+    GrpcClientSettings settings =
+        GrpcClientSettings.connectToServiceAt("my-service", 80, system) // <2>
+          .withTls(false); // <3>
+    counterClient = CounterServiceClient.create(settings, system);
```

Comment:
This is the part I think we should provide some convenience for in the Java SDK. To also solve the client re-use problem we could have a `actionContext().getGrpcClient(CounterService.class, "my-service", 80)`.

Alternatively we could have corresponding `createGrpcClient` in `AkkaServerless` and it has to be passed in to the components in a constructor parameter. That DI style might be better for testing (and should be a familiar style for Java devs).

Reply:
I was think on something like this: 

For external services, user will need to register clients to `AkkaServerless`. We take care of setup and teardown. 

Clients for local components are registered automatically for the user. They don't need to do anything. Seamless experience.

Then, when it comes to use it, user would call: 

```scala
actionContext().clientFor(CounterService.class)
```
`CounterService` could be local or not. At that moment it doesn't matter anymore. 

User could choose to lookup via context or inject via constructor. If they want to inject via constructor, they will first need to register the client (if non-local) and then look it up before registering the component where they want to register it.

Reply:
And ideally, users should not have access to the actor system.

Reply:
It would have to be something like Patrik suggested because the name the service is deployed as is not possible to deduce from the generated client class.

Reply:
We can use `localhost:proxyPort` and avoid going out and back for local components.

Reply:
this is for inter-service though, so two separate deployments - two separate clusters.

Reply:
Updated with an api on Context much like what Patrik suggested, covering both internal Akka Serverless gRpc consumption and external - if the name is present in the `akka.grpc.client` config, that is used (external service) and if not, it is expected to be an internal service over the service mesh mathingy.

Reply:
So, maybe you didn't understand what I was suggesting. 

For inter-services calls, the user need to register a client programmatically or create on the fly based on some config (like in this PR). For that I agree, the user needs to have a name for it for the service discovery. 

For local components, instead of requiring a name to be passed, the user could simply call `clientFor(SomeService.class)` and hit the proxy directly (no service mesh). 

That said, the method `getGrpcClient(Class<T> clientClass, String service)`, could be simplify to `getGrpcClient(Class<T> clientClass)` (I personally prefer `clientFor` though). 

The point being, why bother with a service name for local components? I think we can have a solution where the serviceName and lookup is not required all the time.

For external services, the config key could be based on the FQCN. If we find a config for the class, we use it, the service name should be registered there.

Reply:
For local calls I think it shouldn't be a lookup with class like this at all but rather a class/object that is codegen-generated with clients for each of the components in the service itself. I think it is out of scope for this PR though.

Reply:
The FQCN of the class isn't necessarily the identity of a running external service, multiple servers could implement the same protobuf protocol, so I think it is better to have an explicit name than using the generated client class name, since you anyway have to add config block I don't think there is a lot of mental dev overhead there.